python/pyspark/ml/regression.py

```diff
@@ -743,6 +743,18 @@ def treeWeights(self):
         """Return the weights for each tree"""
         return list(self._call_java("javaTreeWeights"))
 
```

Comment:
I vote for base ```TreeEnsembleModels.trees``` return a list of ```DecisionTreeRegressionModel``` and override this method for ```RandomForestClassificationModel```. We should add docs to clarify it to developers.